### ISSUE TITLE
MySQL Support

### ISSUE THREAD

**Issue by alexrp** [dradovic/MigSharp#4]
Would it be possible to add MySQL support?

Thanks!

**Comment by mediocreguy** [comment]
Hey - I think I've got a MySQL provider put together based on the work you started. However, the testing is failing due to the lack of support in MySQL for renaming a primary key - RenamePrimaryKey gives a NotSupportedException.  Can this test be skipped?  I'm probably missing something...

**Comment by mediocreguy** [comment]
Hey - great package!

I have put together a mostly working provider for MySQL based on your earlier work - the only tests that are failing are in Migration10:

MigSharp.General Error: 0 : An non-recoverable error occurred in Migration '10': 'Temporary default values' while executing Up.
Test 'MigSharp.MySql.NUnit.MySqlIntegrationTests.IntegrationTestsBase.TestMigration1SucceededByAllOtherMigrations' failed:
	The previous query should have failed.
	Integration\Migration10.cs(50,0): at MigSharp.NUnit.Integration.Migration10.<Up>b__0(IRuntimeContext context)
	Core\Commands\CallCommand.cs(23,0): at MigSharp.Core.Commands.CallCommand.<ToSql>d__0.MoveNext()
	at System.Linq.Enumerable.<SelectManyIterator>d__14`2.MoveNext()
	Process\BootstrapMigrationStep.cs(35,0): at MigSharp.Process.BootstrapMigrationStep.Execute(IDbConnection connection, IDbTransaction transaction, MigrationDirection direction, IDbCommandExecutor commandExecutor)
	Process\MigrationStep.cs(67,0): at MigSharp.Process.MigrationStep.Execute(IVersioning versioning)
	Process\MigrationBatch.cs(75,0): at MigSharp.Process.MigrationBatch.ExecuteStep(IMigrationStep step)
	Process\MigrationBatch.cs(65,0): at MigSharp.Process.MigrationBatch.Execute()
	Migrator.cs(84,0): at MigSharp.Migrator.MigrateAll(Assembly assembly, Assembly[] additionalAssemblies)
	Integration\IntegrationTestsBase.cs(237,0): at MigSharp.NUnit.Integration.IntegrationTestsBase.TestMigration1SucceededByAllOtherMigrations()

========================================================================

And Migrate.exe will not run due to:

System.ArgumentException: Unable to find the requested .Net Framework Data Provider.  It may not be installed.

========================================================================

I would be willing to share the code if you are interested - not sure about the next step...

**Comment by dradovic** [comment]
Cool. I'd be happy to add support for MySql!

As for the Migrate.exe problem, I suspect you have to adjust your Migrate.exe.config to include the configuration needed for the MySQL Data Provider.

As for Migration10 that is failing. Mig10 is testing the `HavingTemporaryDefault` feature. This method enables to add a new column to an existing table and specify a temporary default which is used to populate the column for the rows that already exist in that table. After adding and populating the column, this temporary default needs to be discarded again such that INSERTs not specify values for this column fail. This is what is being tested in Mig10 where it says `"The previous query should have failed."`.

Now, it can be that MySQL does not support dropping of default values (SQLite does not as you can see in this migration). In that case, you would need to exclude MySQL from that test in a similar fashion.

In general, feel free to issue a PR once you have the integration tests green.

Thanks a lot!

**Comment by mediocreguy** [comment]
Thanks for the help - I found the problem with Migration10 - MySQL does not throw an exception on INSERT errors unless Strict Mode is enabled - all tests passed. 

I bypassed the command-line tests in a similar manner to the SQLite tests, since MySql support is not built into the .Net framework and would require a reference to MySql.Data to be added to migrate.exe - probably not a good idea.

I've generated a pull request.
